lib/Transforms/Scalar/UnifyAssumes.cc

```diff
@@ -48,48 +56,68 @@ class UnifyAssumesPass : public ModulePass {
 
 char UnifyAssumesPass::ID = 0;
 
-bool isAssumeCall(const CallInst &ci, seahorn::SeaBuiltinsInfo &SBI) {
+bool UnifyAssumesPass::isAssumeCall(const CallInst &ci) {
   using namespace seahorn;
-  switch(SBI.getSeaBuiltinOp(ci)) {
+  switch (m_SBI->getSeaBuiltinOp(ci)) {
   default: return false;
   case SeaBuiltinsOp::ASSUME:
   case SeaBuiltinsOp::ASSUME_NOT:
     return true;
   }
 }
+
+bool UnifyAssumesPass::isAssertCall(const CallInst &ci) {
+  using namespace seahorn;
+  switch (m_SBI->getSeaBuiltinOp(ci)) {
+  default:
+    return false;
+  case SeaBuiltinsOp::ASSERT:
+  case SeaBuiltinsOp::ASSERT_NOT:
+    return true;
+  }
+}
+
 /// Find a function exit basic block.  Assumes that the function has
 /// a unique block with return instruction
-llvm::BasicBlock *findExitBlock(llvm::Function &F) {
+llvm::BasicBlock *UnifyAssumesPass::findExitBlock(llvm::Function &F) {
   for (llvm::BasicBlock &bb : F)
     if (llvm::isa<llvm::ReturnInst>(bb.getTerminator()))
       return &bb;
   return nullptr;
 }
 
-CallInst *findSeahornFail(llvm::Function &F, seahorn::SeaBuiltinsInfo &SBI) {
+CallInst *UnifyAssumesPass::findSeahornFail(llvm::Function &F) {
   for (auto &inst : instructions(F)) {
     if (auto *CI = dyn_cast<CallInst>(&inst)) {
-      if (SBI.getSeaBuiltinOp(*CI) == seahorn::SeaBuiltinsOp::FAIL) return CI;
+      if (m_SBI->getSeaBuiltinOp(*CI) == seahorn::SeaBuiltinsOp::FAIL)
+        return CI;
     }
   }
   return nullptr;
 }
 
 bool UnifyAssumesPass::runOnFunction(Function &F) {
   Module &M = *F.getParent();
-  auto &SBI = getAnalysis<seahorn::SeaBuiltinsInfoWrapperPass>().getSBI();
-  auto *assumeFn = SBI.mkSeaBuiltinFn(seahorn::SeaBuiltinsOp::ASSUME, M);
+  m_SBI = &getAnalysis<seahorn::SeaBuiltinsInfoWrapperPass>().getSBI();
+  assert(m_SBI);
+  auto *assumeFn = m_SBI->mkSeaBuiltinFn(seahorn::SeaBuiltinsOp::ASSUME, M);
 
   SmallVector<CallInst *, 16> assumes;
+  SmallVector<CallInst *, 16> asserts;
   for (auto &inst : instructions(F)) {
     if (auto *ci = dyn_cast<CallInst>(&inst)) {
-      if (isAssumeCall(*ci, SBI))
+      if (isAssumeCall(*ci)) {
         assumes.push_back(ci);
+      } else if (isAssertCall(*ci)) {
+        asserts.push_back(ci);
+      }
     }
   }
-  // -- nothing to unify if there is only one assume already
-  if (assumes.size() <= 1)
+  if (assumes.size() < 1) {
```

Comment:
remove. We agreed it is not a warning. We will deal with `verifier.assert` in opsem

Reply:
done

Reply:
`assumes.size() < 1` is the same as `assumes.size() == 0` same as `assumes.empty()`

Reply:
fixed